include/kframework/lemmas/lemmas.k

```diff
@@ -28,14 +28,18 @@ module LEMMAS [symbolic]
     rule         chop(_V) <Int pow256 => true        [simplification]
 
     rule X *Int Y <Int pow256 => true requires Y <=Int maxUInt256 /Int X [simplification]
-    rule #if B #then C +Int C1 #else C +Int C2 #fi => C +Int #if B #then C1 #else C2 #fi [simplification] 
+    rule #if B #then C +Int C1 #else C +Int C2 #fi => C +Int #if B #then C1 #else C2 #fi [simplification]
 
   // ########################
   // Set Reasoning
   // ########################
 
-    rule X in (SetItem(Y) _   ) => true      requires X  ==Int Y [simplification]
-    rule X in (SetItem(Y) REST) => X in REST requires X =/=Int Y [simplification]
+    rule X in SetItem ( Y ) S:Set => ( X ==K Y andBool notBool ( X in S ) ) orBool ( X =/=K Y andBool X in S ) [simplification]
+    rule X in S:Set SetItem ( Y ) => ( X ==K Y andBool notBool ( X in S ) ) orBool ( X =/=K Y andBool X in S ) [simplification]
```

Comment:
Is the use of `==K` instead of `==Int` to make the lemma more general? Are those operators always identical when applied to Ints?

Reply:
Yes, it will work for arbitrary sets, including sets of integers, in which case the backend may automatically convert `==K` to `==Int`, I believe.